app/src/components/map/MapFeatureList.tsx

```diff
@@ -48,35 +58,81 @@ const MapFeatureList: React.FC<MapFeatureListProps> = (props) => {
     activeFeatureState[1](null);
   };
 
+  const deleteListItem = (index: number) => {
```

Comment:
This does work as a custom function, but an ArrayHelper makes ait alot simpler in code. #[Check ](https://github.com/bcgov/nert-restoration-tracker/blob/0c9ad9edad08964ed90a62033a9dc348b64e81fd/app/src/features/projects/components/ProjectObjectivesForm.tsx)
```
<FieldArray
        name="objective.objectives"
        render={(arrayHelpers: any) => (
...
 <IconButton
                              color="primary"
                              data-testid="delete-icon"
                              aria-label="remove objective"
                              onClick={() => arrayHelpers.remove(index)}
                              edge="end"
                              size="large">
                              <Icon path={mdiTrashCanOutline} size={1} />
                            </IconButton>

```

Reply:
What is this <FieldArray> you speak of hahaha :laughing: .
I'm looking into the documentation and it looks pretty cool. The only thing is I can't see a way of incorporating the custom Mask logic. Looks great for removing the entry from the Formik data... However, I don't see a way to manipulate other data... such as the mask state array.

Reply:
In the same way you can remove an array object you can add one. Such as 
``` <Button
                disabled={
                  !values.objective.objectives[
                    values.objective.objectives.length - 1
                  ].objective.trim() || values.objective.objectives.length >= 5
                }
                type="button"
                variant="outlined"
                color="primary"
                aria-label="add objective"
                startIcon={<Icon path={mdiPlus} size={1}></Icon>}
                onClick={() => arrayHelpers.push(ProjectObjectivesFormArrayItemInitialValues)}>
                Add New Objective
              </Button>```
              
Where on click the its pushing a new object onto the array. So similarly you can add a custom function to this onclick to handle any gets or small edits you may want to do. But simplifies the adding and removing of items on the list.

Reply:
K... I'll giv'r a go

Reply:
So I've got the Array Helper to wrap the feature list. Having lots of issues with the map features getting out of sync with the list items.